build-tools/packages/build-infrastructure/src/versions.ts

```diff
@@ -0,0 +1,33 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import * as semver from "semver";
+
+import { updatePackageJsonFileAsync } from "./packageJsonUtils.js";
+import type { IPackage, PackageJson } from "./types.js";
+
+/**
+ * Sets the version of a group of packages.
+ *
+ * Note that any loaded objects such as an IFluidRepo instance may need to be reloaded after calling this function.
```

Comment:
How will the user know if such a reload is needed or not? What does the error mode look like?

Reply:
Also nit:
```suggestion
 * Note that any loaded objects such as an {@link IFluidRepo} instance may need to be reloaded after calling this function.
```

Slightly more likely to get auto-fixed by refactoring tools if the type gets renamed, etc.

Reply:
> How will the user know if such a reload is needed or not? What does the error mode look like?

No error mode - the in-memory packages may be outdated. I toyed around with taking the fluid repo instance, but that seemed weird for a low-level function like this. I think most of the validation/checking will be done in higher-level functions that build on top of this one.

Reply:
Turns out I can avoid this confusion by calling reload on the packages after their package.json is updated. So now no
reload is needed.